add some basic shape_env info to tlparse

bdhirsh opened the pull request:
It seems generally useful to add stuff like shape_env hints on a given compile into tlparse, especially since hints are used by the partitioner / inductor and can affect compile times and performance (see [doc](https://docs.google.com/document/d/1SyOGPKKVQAmnEY-LkGbcVj1H6TTMtsXu0DRVVQJqCGw/edit?tab=t.0#bookmark=id.bc18vdapcxwf))

Example tlparse: https://manifold.edge.x2p.facebook.net/v0/read/tree/logs/.tmpuO2Pey/index.html?bucketName=tlparse_reports&apiKey=tlparse_reports-key&withPayload=1&timeoutMsec=100

From this code:
```
import torch


@torch.compile
def f(x, y):
    if x.shape[0] > y.shape[0]:
        return x.sum() + y.sum()
    else:
        return x.sum() - y.sum()


x = torch.randn(5, 6)
y = torch.randn(6, 3)
out = f(x, y)

x = torch.randn(7, 3)
y = torch.randn(8, 3)
out = f(x, y)
```

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #144438
* #144097



cc @ezyang @SherlockNoMad @EikanWang @jgong5 @wenzhe-nrv @voznesenskym @penguinwu @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @jiayisunx @chenyang78 @kadeng @chauhang @amjames

bdhirsh committed:
add some basic shape_env info to tlparse

[ghstack-poisoned]

jamesjwu commented on torch/_dynamo/output_graph.py:
```diff
@@ -1381,6 +1381,16 @@ def compile_and_call_fx_graph(self, tx, rv, root, replaced_outputs):
             with self.restore_global_state():
                 compiled_fn = self.call_user_compiler(gm)
 
+            torch._logging.trace_structured(
```
Any reason not to use encoding: json here? That way you could return json.dumps() and have slightly more structured data.
 
Example usage:
https://github.com/pytorch/pytorch/blob/b0be30dd7912797dcdbe01c2b1186f45482813ff/torch/_functorch/_aot_autograd/autograd_cache.py#L362-L369

bdhirsh replied:
I was just cargo-culting, this definitely seems better - will update!

bdhirsh committed:
Update on "add some basic shape_env info to tlparse"

It seems generally useful to add stuff like shape_env hints on a given compile into tlparse, especially since hints are used by the partitioner / inductor and can affect compile times and performance (see [doc](https://docs.google.com/document/d/1SyOGPKKVQAmnEY-LkGbcVj1H6TTMtsXu0DRVVQJqCGw/edit?tab=t.0#bookmark=id.bc18vdapcxwf))

Example tlparse: https://manifold.edge.x2p.facebook.net/v0/read/tree/logs/.tmpuO2Pey/index.html?bucketName=tlparse_reports&apiKey=tlparse_reports-key&withPayload=1&timeoutMsec=100

From this code:
```
import torch


torch.compile
def f(x, y):
    if x.shape[0] > y.shape[0]:
        return x.sum() + y.sum()
    else:
        return x.sum() - y.sum()


x = torch.randn(5, 6)
y = torch.randn(6, 3)
out = f(x, y)

x = torch.randn(7, 3)
y = torch.randn(8, 3)
out = f(x, y)
```




cc ezyang SherlockNoMad EikanWang jgong5 wenzhe-nrv voznesenskym penguinwu Guobing-Chen XiaobingSuper zhuhaozhe blzheng jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

bdhirsh committed:
Update on "add some basic shape_env info to tlparse"

It seems generally useful to add stuff like shape_env hints on a given compile into tlparse, especially since hints are used by the partitioner / inductor and can affect compile times and performance (see [doc](https://docs.google.com/document/d/1SyOGPKKVQAmnEY-LkGbcVj1H6TTMtsXu0DRVVQJqCGw/edit?tab=t.0#bookmark=id.bc18vdapcxwf))

Example tlparse: https://manifold.edge.x2p.facebook.net/v0/read/tree/logs/.tmpuO2Pey/index.html?bucketName=tlparse_reports&apiKey=tlparse_reports-key&withPayload=1&timeoutMsec=100

From this code:
```
import torch


torch.compile
def f(x, y):
    if x.shape[0] > y.shape[0]:
        return x.sum() + y.sum()
    else:
        return x.sum() - y.sum()


x = torch.randn(5, 6)
y = torch.randn(6, 3)
out = f(x, y)

x = torch.randn(7, 3)
y = torch.randn(8, 3)
out = f(x, y)
```




cc ezyang SherlockNoMad EikanWang jgong5 wenzhe-nrv voznesenskym penguinwu Guobing-Chen XiaobingSuper zhuhaozhe blzheng jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

bdhirsh committed:
Update on "add some basic shape_env info to tlparse"

It seems generally useful to add stuff like shape_env hints on a given compile into tlparse, especially since hints are used by the partitioner / inductor and can affect compile times and performance (see [doc](https://docs.google.com/document/d/1SyOGPKKVQAmnEY-LkGbcVj1H6TTMtsXu0DRVVQJqCGw/edit?tab=t.0#bookmark=id.bc18vdapcxwf))

Example tlparse: https://manifold.edge.x2p.facebook.net/v0/read/tree/logs/.tmpuO2Pey/index.html?bucketName=tlparse_reports&apiKey=tlparse_reports-key&withPayload=1&timeoutMsec=100

From this code:
```
import torch


torch.compile
def f(x, y):
    if x.shape[0] > y.shape[0]:
        return x.sum() + y.sum()
    else:
        return x.sum() - y.sum()


x = torch.randn(5, 6)
y = torch.randn(6, 3)
out = f(x, y)

x = torch.randn(7, 3)
y = torch.randn(8, 3)
out = f(x, y)
```




cc ezyang SherlockNoMad EikanWang jgong5 wenzhe-nrv voznesenskym penguinwu Guobing-Chen XiaobingSuper zhuhaozhe blzheng jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

bdhirsh committed:
Update on "add some basic shape_env info to tlparse"

It seems generally useful to add stuff like shape_env hints on a given compile into tlparse, especially since hints are used by the partitioner / inductor and can affect compile times and performance (see [doc](https://docs.google.com/document/d/1SyOGPKKVQAmnEY-LkGbcVj1H6TTMtsXu0DRVVQJqCGw/edit?tab=t.0#bookmark=id.bc18vdapcxwf))

Example tlparse: https://manifold.edge.x2p.facebook.net/v0/read/tree/logs/.tmpuO2Pey/index.html?bucketName=tlparse_reports&apiKey=tlparse_reports-key&withPayload=1&timeoutMsec=100

From this code:
```
import torch


torch.compile
def f(x, y):
    if x.shape[0] > y.shape[0]:
        return x.sum() + y.sum()
    else:
        return x.sum() - y.sum()


x = torch.randn(5, 6)
y = torch.randn(6, 3)
out = f(x, y)

x = torch.randn(7, 3)
y = torch.randn(8, 3)
out = f(x, y)
```




cc ezyang SherlockNoMad EikanWang jgong5 wenzhe-nrv voznesenskym penguinwu Guobing-Chen XiaobingSuper zhuhaozhe blzheng jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

bdhirsh committed:
Update on "add some basic shape_env info to tlparse"

It seems generally useful to add stuff like shape_env hints on a given compile into tlparse, especially since hints are used by the partitioner / inductor and can affect compile times and performance (see [doc](https://docs.google.com/document/d/1SyOGPKKVQAmnEY-LkGbcVj1H6TTMtsXu0DRVVQJqCGw/edit?tab=t.0#bookmark=id.bc18vdapcxwf))

Example tlparse: https://manifold.edge.x2p.facebook.net/v0/read/tree/logs/.tmpuO2Pey/index.html?bucketName=tlparse_reports&apiKey=tlparse_reports-key&withPayload=1&timeoutMsec=100

From this code:
```
import torch


torch.compile
def f(x, y):
    if x.shape[0] > y.shape[0]:
        return x.sum() + y.sum()
    else:
        return x.sum() - y.sum()


x = torch.randn(5, 6)
y = torch.randn(6, 3)
out = f(x, y)

x = torch.randn(7, 3)
y = torch.randn(8, 3)
out = f(x, y)
```




cc ezyang SherlockNoMad EikanWang jgong5 wenzhe-nrv voznesenskym penguinwu Guobing-Chen XiaobingSuper zhuhaozhe blzheng jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

bdhirsh committed:
Update on "add some basic shape_env info to tlparse"

It seems generally useful to add stuff like shape_env hints on a given compile into tlparse, especially since hints are used by the partitioner / inductor and can affect compile times and performance (see [doc](https://docs.google.com/document/d/1SyOGPKKVQAmnEY-LkGbcVj1H6TTMtsXu0DRVVQJqCGw/edit?tab=t.0#bookmark=id.bc18vdapcxwf))

Example tlparse: https://manifold.edge.x2p.facebook.net/v0/read/tree/logs/.tmpuO2Pey/index.html?bucketName=tlparse_reports&apiKey=tlparse_reports-key&withPayload=1&timeoutMsec=100

From this code:
```
import torch


torch.compile
def f(x, y):
    if x.shape[0] > y.shape[0]:
        return x.sum() + y.sum()
    else:
        return x.sum() - y.sum()


x = torch.randn(5, 6)
y = torch.randn(6, 3)
out = f(x, y)

x = torch.randn(7, 3)
y = torch.randn(8, 3)
out = f(x, y)
```




cc ezyang SherlockNoMad EikanWang jgong5 wenzhe-nrv voznesenskym penguinwu Guobing-Chen XiaobingSuper zhuhaozhe blzheng jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

bdhirsh committed:
Update on "add some basic shape_env info to tlparse"

It seems generally useful to add stuff like shape_env hints on a given compile into tlparse, especially since hints are used by the partitioner / inductor and can affect compile times and performance (see [doc](https://docs.google.com/document/d/1SyOGPKKVQAmnEY-LkGbcVj1H6TTMtsXu0DRVVQJqCGw/edit?tab=t.0#bookmark=id.bc18vdapcxwf))

Example tlparse: https://manifold.edge.x2p.facebook.net/v0/read/tree/logs/.tmpuO2Pey/index.html?bucketName=tlparse_reports&apiKey=tlparse_reports-key&withPayload=1&timeoutMsec=100

From this code:
```
import torch


torch.compile
def f(x, y):
    if x.shape[0] > y.shape[0]:
        return x.sum() + y.sum()
    else:
        return x.sum() - y.sum()


x = torch.randn(5, 6)
y = torch.randn(6, 3)
out = f(x, y)

x = torch.randn(7, 3)
y = torch.randn(8, 3)
out = f(x, y)
```




cc ezyang SherlockNoMad EikanWang jgong5 wenzhe-nrv voznesenskym penguinwu Guobing-Chen XiaobingSuper zhuhaozhe blzheng jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

bdhirsh committed:
Update on "add some basic shape_env info to tlparse"

It seems generally useful to add stuff like shape_env hints on a given compile into tlparse, especially since hints are used by the partitioner / inductor and can affect compile times and performance (see [doc](https://docs.google.com/document/d/1SyOGPKKVQAmnEY-LkGbcVj1H6TTMtsXu0DRVVQJqCGw/edit?tab=t.0#bookmark=id.bc18vdapcxwf))

Example tlparse: https://manifold.edge.x2p.facebook.net/v0/read/tree/logs/.tmpuO2Pey/index.html?bucketName=tlparse_reports&apiKey=tlparse_reports-key&withPayload=1&timeoutMsec=100

From this code:
```
import torch


torch.compile
def f(x, y):
    if x.shape[0] > y.shape[0]:
        return x.sum() + y.sum()
    else:
        return x.sum() - y.sum()


x = torch.randn(5, 6)
y = torch.randn(6, 3)
out = f(x, y)

x = torch.randn(7, 3)
y = torch.randn(8, 3)
out = f(x, y)
```




cc ezyang SherlockNoMad EikanWang jgong5 wenzhe-nrv voznesenskym penguinwu Guobing-Chen XiaobingSuper zhuhaozhe blzheng jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]
